File: client/src/routes/utils/schedule.ts
import { Day } from 'enums';

const toDay = (day: string): Day => {
  switch (day) {
    case 'Monday':
      return Day.Monday;
    case Day.Tuesday.toString():
      return Day.Tuesday;
    case Day.Wednesday.toString():
      return Day.Wednesday;
    case Day.Thursday.toString():
      return Day.Thursday;
    case Day.Friday.toString():
      return Day.Friday;
    case Day.Saturday.toString():
      return Day.Saturday;
    case Day.Sunday.toString():
      return Day.Sunday;
    default:
      throw new Error(`toDay recieved invalid value '${day}' for day.`);
  }
};

export {
  toDay,
};

File: server/src/main/kotlin/com/marisarichmond/honeydew/models/Schedule.kt
package com.marisarichmond.honeydew.models

import java.util.UUID
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.ManyToOne
import javax.persistence.Table

enum class Day {
    Monday,
    Tuesday,
    Wednesday,
    Thursday,
    Friday,
    Saturday,
    Sunday,
}

@Entity
@Table(name = "schedule")
data class Schedule(
    @Id
    override val id: UUID = UUID.randomUUID(),
    @ManyToOne
    @JoinColumn(name = "user_id")
    val user: User,

    val day: Day,
    val isActive: Boolean = false,
    val timeInMinutes: Long = 0,
) : Base(id) {
    override fun toString() = this::class.simpleName + "(" + listOf(
        "id = $id",
        "day = $day",
        "isActive = $isActive",
        "timeInMinutes = $timeInMinutes"
    ).joinToString(", ") + ")"
}
